rs/consensus/idkg/src/pre_signer.rs

```diff
@@ -390,143 +390,147 @@ impl IDkgPreSignerImpl {
+                                self.metrics.pre_sign_errors_inc("duplicate_support");
+                                return Some(IDkgChangeAction::HandleInvalid(
+                                    id,
+                                    format!("Duplicate support: {support}"),
+                                ))
+                            }
+
+                            if support.transcript_id != dealing.transcript_id
+                                || support.dealer_id != signed_dealing.dealer_id()
+                            {
+                                // Meta data mismatch
+                                self.metrics
+                                    .pre_sign_errors_inc("support_meta_data_mismatch");
+                                return Some(IDkgChangeAction::HandleInvalid(
+                                    id,
+                                    format!(
+                                        "Support meta data mismatch: expected = {:?}/{:?}, \
+                                            received = {:?}/{:?}",
+                                        support.transcript_id,
+                                        support.dealer_id,
+                                        dealing.transcript_id,
+                                        signed_dealing.dealer_id()
+                                    ),
+                                ))
+                            }
+
+                            let Some(transcript_params) = self.resolve_ref(
+                                transcript_params_ref,
+                                block_reader,
+                                "validate_dealing_support",
+                            ) else {
+                                return Some(IDkgChangeAction::HandleInvalid(
+                                    id.clone(),
+                                    format!("Failed to translate transcript_params_ref: {}", support),
+                                ));
+                            };
+
                             let action = self.crypto_verify_dealing_support(
-                                id,
+                                &id,
                                 &transcript_params,
                                 signed_dealing,
                                 support,
                                 idkg_pool.stats(),
                             );
-                            if let Some(IDkgChangeAction::MoveToValidated(_)) = action {
-                                if let Some(signers) = maybe_signers {
-                                    signers.insert(signer);
-                                } else {
-                                    valid_dealing_supports.insert(key, BTreeSet::from([signer]));
+                            if let Some(IDkgChangeAction::MoveToValidated(msg)) = &action {
+                                let mut valid_dealing_supports = self.validated_dealing_supports.write().unwrap();
```

Comment:
Maybe put a shorter version of what you wrote in the PR description here as a comment